src/Features/Core/Portable/ExtractMethod/MethodExtractor.Analyzer.cs

```diff
@@ -86,8 +86,14 @@ public async Task<AnalyzerResult> AnalyzeAsync()
                     Contract.ThrowIfFalse(unused.Count == 0);
                 }
 
-                // check whether instance member is used inside of the selection
-                var instanceMemberIsUsed = IsInstanceMemberUsedInSelectedCode(dataFlowAnalysisData);
+                var thisParameterBeingRead = dataFlowAnalysisData.ReadInside.FirstOrDefault(s => IsThisParameter(s)) as IParameterSymbol;
+                var isThisParameterWritten = dataFlowAnalysisData.WrittenInside.Any(s => IsThisParameter(s));
+
+                var instanceMemberIsUsed = thisParameterBeingRead is object || isThisParameterWritten;
```

Comment:
why `isObject` isntead of `!= null`?  `is object` really makes it look like a type-test is occurring, which feels strange given that you don't actually care about hte type, but really are ensuring it's not null.  As such `!= null` feels much clearer as to what you're trying to do.

Reply:
I don't have a strong position about `expr is object` versus `expr != null`, but I think Roslyn team should write down a guideline about it and bring this code into compliance with whatever that guideline is. I have tended to see `is object` or `is { }` used in our codebase instead of `!= null` and I agree that the latter is easier to read.

Reply:
for the IDE, where these are just interfaces, and we almost *never* overload `==` i would just have thse be null checks.   :)